litgpt/generate/base.py

```diff
@@ -194,6 +194,120 @@ def generate_fn(
         yield from tokens[yielded_idx:]
 
 
+# TODO: Make include_eos work.
+# TODO: Rewrite unbatched generate_fn to use batched_generate_fn.
+@torch.inference_mode()
+def batched_generate_fn(
+    model: GPT,
+    prompts: torch.Tensor,
+    max_returned_tokens: int,
+    *,
+    sample_args: Union[list[dict], dict],
+    stop_tokens: Tuple[List[int], ...] = (),
+    include_prompt: bool,
+    include_eos: bool,
+) -> Iterator[list[Union[torch.Tensor, None]]]:
+
```

Comment:
Given the complexity of the this, would it make sense to have a docstring describing the args and what their relation needs to be?

Reply:
Yeah, agreed. And also for `generate_fn()`.

The way that we've been thinking about it until this point is that, strictly speaking, anything that isn't accessible from a script or in `litgpt.api` isn't publicly facing. It's clear though that a lot of it is. We still have to go back and document a lot of that. Documenting these would be a good start.

Reply:
Done